2_process.R

```diff
@@ -62,6 +69,26 @@ p2_targets_list <- list(
     p2_site_list_nontidal_csv,
     create_site_list_nontidal(p2_wqp_SC_filtered,p1_nwis_sites,p1_daily_data,p1_inst_data,
                               hucs=drb_huc8s,crs_out="NAD83",p2_sites_w_segs,"2_process/out/DRB_SC_sitelist_nontidal.csv"),
-    format = "file")
-
+    format = "file"),
+  
+  # # target for NADP 
+  tar_target(p2_NADP_Data,
+             lapply(list.files(path = p1_NADP_data_unzipped, full.names = T), function(x) read.csv(x, sep = ',') %>%
+                    # select only cols starting with cat and COMID co
+                    select(COMID | starts_with('CAT')) %>%
+                    # take only COMIDS in drb
+                    filter(COMID %in% p2_drb_comids_all_tribs$comid) %>%
+                    # add year col to ID each dataset - using regex to extract the year between NADP_ and _CONUS
+                    mutate(Year = str_extract_all(x, "(?<=unzipped/NADP_).+(?=_CONUS.txt)")) %>%
+                    # remove year in col name to have all colnames equal across datasets
+                    setNames(gsub('_\\d{4}', '', names(.)))) %>%
+               # rbind the list of cleaned dfs
+      do.call(rbind, .)
+  )
```

Comment:
Notes here: 
* Running lapply - it takes some time to process as each CONUS annual NADP dataset is large (nrow ~= 3 million). Would purr::map() work better? 
* This is a large dplyr code chunk for the 2_process.R file. Chose to work on it directly in the file to produce it, but I can place into a separate function if that is cleaner. 
* regex heavily used to extract items from cells. Open to suggestions on non-regex str extraction methods.

Reply:
I'm not sure about speed for lapply vs. map, but it can definitely be parallelized over `p1_NADP_data_unzipped`